mistralrs-core/src/engine/mod.rs

```diff
@@ -452,28 +457,48 @@ impl Engine {
         let topp = request.sampling_params.top_p.unwrap_or(1.0);
         let num_hidden_layers = get_mut_arcmutex!(self.pipeline).num_hidden_layers();
         let tokenizer = get_mut_arcmutex!(self.pipeline).tokenizer();
+        let tok_trie = get_mut_arcmutex!(self.pipeline).tok_trie();
+
+        let (stop_toks, stop_strings) = match request.sampling_params.stop_toks {
+            None => (vec![], vec![]),
+            Some(StopTokens::Ids(ref i)) => {
+                for id in i {
+                    // We can't use ` ` (space) as a stop token because other tokens like ` moon` start with a space.
+                    if tok_trie.has_extensions(tok_trie.token(*id)) {
+                        request
+                            .response
+                            .send(Response::ValidationError(
+                                format!("Stop token {:?} is also a prefix of other tokens and cannot be used as a stop token.", tok_trie.token_str(*id)).into(),
+                            ))
+                            .unwrap();
+                        return;
```

Comment:
I re-used the validation for valid stop tokens here.

Reply:
Sounds good.